### PR TITLE
fix: Publish trusted-sync to GHCR

### LINKED ISSUE #311: chore(release): Move trusted sync image to `ghcr` registry
## Overview

Currently, we release the `trusted-sync` image to the dockerhub registry. As our other `build` images are released to `ghcr`, we should standardize on a single container registry.

An example of a `ghcr` release workflow can be found @ https://github.com/ethereum-optimism/kona/blob/main/.github/workflows/cross_docker.yaml, which can be referenced to modify the existing `trusted-sync` release workflow.

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Publish `trusted-sync` to ghcr.io.

**Metadata**

Fixes #311

### CODE REVIEW

**Comment by refcell** [comment]
* **#312** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/312?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @refcell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/cross_docker.yaml [hunk 1]
```diff
@@ -35,3 +38,29 @@ jobs:
...
+      contents: read
+      packages: write
+    env:
+      IMAGE_NAME: ${{ github.repository }}/trusted-sync
+    steps:
+      - uses actions/checkout@v4
+      - uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
```
Comment (clabby) [inline .github/workflows/cross_docker.yaml:53]: Think this needs a `password` field
```suggestion
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

**Review by clabby — CHANGES_REQUESTED** [review]
Can we move the `publish-trusted-sync` job back into the other workflow made in the previous PR?

Reason being, when we run the `cross_docker.yaml` release, it'll publish both the trusted sync and build images atm.

**Review by clabby — APPROVED** [review]

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  publish-ghcr:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    env:
+      IMAGE_NAME: ${{ github.repository }}/trusted-sync
+    steps:
+      - uses actions/checkout@v4
```
Comment (clabby) [inline .github/workflows/publish.yaml:15]: Syntax error here
```suggestion
      - uses: actions/checkout@v4
```

**Pull request merged**